ISISComputingGroup/ibex_gui
# Ticket5811 table bug fixing

### Description of work

- Correct how we update actions in the script generator actions table so we don't lose and reapply focus
- Fix some minor bugs discovered whilst playing around

### Ticket

https://github.com/ISISComputingGroup/IBEX/issues/5811

### Acceptance criteria

#### To test:

- Add lots of rows very quickly and then try editing and tabbing between cells whilst the script generator validates and calculates an estimated time for all the actions
- Confirm that general actions still work such as changing script definitions, adding, removing, moving actions up and down, saving and reloading parameters 

### Unit tests

*Give an overview of unit tests you have added or modified, if applicable. The aim is provide information to help the reviewer*

### System tests

*Mention any automated tests or manual tests that you have added or modified, if applicable. The aim is provide information to help the reviewer*

### Documentation
*Highlight and provide a link to any additions or changes to the documentation, if applicable. The aim is provide information to help the reviewer*

---

#### Code Review


### Final Steps
